code/modules/station_goals/bsa.dm

```diff
@@ -220,8 +223,18 @@
 	reload()
 
 /obj/machinery/bsa/full/proc/reload()
-	use_power(power_used_per_shot)
-	COOLDOWN_START(src, firing_cooldown, reload_cooldown_time)
+	if(machine_powernet?.powernet_apc?.cell?.charge KJ >= energy_used_per_shot)
+		try_reload = FALSE
+		use_power(energy_used_per_shot)
+		COOLDOWN_START(src, firing_cooldown, reload_cooldown_time)
+	else
+		firing_cooldown = reload_cooldown_time
```

Comment:
Probably don't need to reset the firing cooldown here since it'll always end up past the cooldown if we're here, right?

Reply:
yeah, that's redundant. removed it.